pkgs/development/compilers/gambit/bootstrap.nix

```diff
@@ -1,16 +1,16 @@
-{ stdenv, fetchurl, autoconf, ... }:
+{ stdenv, fetchurl, autoconf, git, ... }:
 
 stdenv.mkDerivation rec {
   name    = "gambit-bootstrap-${version}";
-  version = "4.8.9";
-  tarball_version = "v4_8_9";
+  version = "4.9.1";
+  tarball_version = "v4_9_1";
 
   src = fetchurl {
-    url    = "http://www.iro.umontreal.ca/~gambit/download/gambit/v4.8/source/gambit-${tarball_version}-devel.tgz";
-    sha256 = "b7f86c794711792ca556ce41f8bc7043dffc395c01bb6d8d119bc2f454f89fbf";
+    url    = "http://www.iro.umontreal.ca/~gambit/download/gambit/v4.9/source/gambit-${tarball_version}-devel.tgz";
+    sha256 = "10kzv568gimp9nzh5xw0h01vw50wi68z3awfp9ibqrpq2l0n7mw7";
   };
 
-  buildInputs = [ autoconf ];
+  buildInputs = [ autoconf git ];
```

Comment:
How does it make use of git?

Reply:
Somewhere in the Makefile it uses git. Maybe not anymore, that we use fetchFromGitHub instead of fetchgit: that might be predicated on directory `.git` existing.

Reply:
Indeed the `lib/makefile` has a rule `deselect-gen-for-commit-post` that executes:
```
if [ -e ../.git ]; then git update-index --skip-worktree _kernel.c _system.c _num.c _std.c _eval.c _io.c _nonstd.c _thread.c _repl.c _gambit.c; fi
```
and the bootstrap tarball does contain the entire git tree (I'll ask the upstream author whether that's on purpose).

Reply:
Also `fetchgit` does not contain a `.git` directory.

Reply:
Can we remove it then?

Reply:
For the bootstrap image, I use the recommended upstream tarball, that contains a `.git` directory and requires the `git` executable.

I could use `fetchgit` or `fetchFromGitHub` instead if you believe that's a better approach, and then remove the dependency on `git`.

Reply:
ok. we can leave it then as is. `git` should be in `nativeBuildInputs` then.

Reply:
Oh, you mean put both of them in `nativeBuildInputs` instead of `buildInputs` ?

Reply:
Or should `git` be in `nativeBuildInputs` and `autoconf` be in `depsBuildInputs` ? I admit I don't understand the criteria for which inputs is which. Furthermore, I believe that cross-compilation of these particular packages may not be working, anyway.

Reply:
Is there a flag I should set somehow to tell nix that a cross-build is probably doomed to fail?

Reply:
It is ok, when the current expression does not support it, but it looks like the project itself support it.

Reply:
My current expression definitely doesn't support cross-compilation. I'm not even sure where to start from to support it, or how to test it. Probably with x86 vs x86_64, but even then I'm not sure how to do it.

Reply:
I'll mark as resolved: I don't currently have the time or skills to work on a recipe that supports cross-compilation, and I don't believe anyone else has the interest.

Reply:
We are trying to move it at least new code in a direction that makes it easier to reason about cross compilation at some point.